Reorganize the Modules chapter of the manual. (#38271)
JuliaLang/julia: doc/src/manual/modules.md

2. Modules have facilities for detailed namespace management: each defines a set of symbols it
   `export`s, and can import symbols from other modules with `using` and `import` (we explain these below).

3. Modules can be precompiled for faster loading, and contain code for runtime initialization.

Typically, in larger Julia packages you will see module code organized into files, eg

```julia
module SomeModule

# export, using, import statements are usually here; we discuss these below

include("file1.jl")
include("file2.jl")

end
```

Files and file names are mostly unrelated to modules; modules are associated only with module
expressions. One can have multiple files per module, and multiple modules per file, `include`
behaves as if the contents of the source file were evaluated in its place. In this chapter, we use
short and simplified examples, so we won't use `include`.

The recommended style is not to indent the body of the module, since that would typically lead to
whole files being indented. Also, it is common to use `MixedCase` for module names (just like

> types), and use the plural form if applicable, especially if the module contains a similarly named

Reviewer: In https://github.com/JuliaLang/julia/blob/4562ecef9939a8f132c706213fd628f5d0978c55/doc/src/manual/variables.md#stylistic-conventions this is referred to as "upper camel case"

Author: Done, thanks!

# [Modules](@id modules)

> Modules in Julia help organizing code into coherent units. They are delimited syntactically inside

Reviewer 2: ```suggestion
Modules in Julia help organize code into coherent units. They are delimited syntactically inside
```

> `module Name ... end`, and have the following features:

Reviewer 2: ```suggestion
`module NameOfModule ... end`, and have the following features:
```

1. Modules are separate namespaces, each introducing a new global scope. This is useful, because it

>    allows functions and global variables to have the same name in different modules without being in
>    conflict.

Reviewer 2: ```suggestion
   allows the same name to be used for different functions or global variables without conflict, as long as they are in separate modules.
```
